packages/binding-modbus/src/modbus-client.ts

```diff
@@ -212,22 +212,26 @@ export default class ModbusClient implements ProtocolClient {
         return endianness;
     }
 
-    private overrideFormFromURLPath(input: ModbusForm) {
+    // This generates a form with url content based on the uri scheme
+    // Ideally, more code should be refactored to use uri only
+    private generateFormFromURLPath(input: ModbusForm) {
```

Comment:
I'm ok with finding a better name for this function. However, `generate` does not really explain the fact that the function actually modifies the form in input with the values taken from the URI. We can clone the form input and make it immutable to better fit the description.

Reply:
Good point. We are overriding the form but more that we are adding terms into it whereas before the URI values would be prioritized (no code change though). How about `addFormElementsFromURLPath`

Reply:
I like the idea of not changing the input form and rather clone and update. Doing so the term `generate` with reporting back the update is fine I think

Reply:
This is now done. Can we resolve this thread and merge the PR?